### DISCUSSION TITLE
CLient side env finder using prefix NEXT_PUBLIC_ not working as expected

### DISCUSSION THREAD

**Discussion by mike-rambil** [vercel/next.js#66246] [App Router] [answered]
I have the log properly in server side but the env is not accessible in the browser!

// clientsideutils.ts
'use client'

import { fetchToken } from './fetchToken_backend'

export async function envFinderClient(FOUND_ENV: string): Promise<string> {
  console.log('🚀 ~ FOUND_ENV:', FOUND_ENV)
  const foundENV = process.env[FOUND_ENV]
  console.log('🚀 ~ foundENV:', foundENV)
  if (foundENV) {
    return foundENV
  } else {
    throw new Error(`${FOUND_ENV} environment variable is not set`)
  }
}

//#region Barrel Events & Reasons Storage

export async function getBarrel_EventReasons() {
  // Cached API
  try {
    const token = await fetchToken()
    const backendURL = await envFinderClient('NEXT_PUBLIC_BACKEND_WMS_SERVER')
    const response = await fetch(`${backendURL}/api/eventReasons`, {
      method: 'GET',
      headers: {
        'Content-Type': 'application/json',
        Accept: 'application/json',
        Authorization: `Bearer ${token}`
      }
    })

    const responseData = await response.json()
    // console.log('🚀 ~ BARREL REASONS:', responseData)
    if (typeof window !== 'undefined' && window.sessionStorage) {
      // Store data in local storage instead of session storage
      sessionStorage.setItem('WMSeventReasons', JSON.stringify(responseData))
    }
    return
  } catch (error) {
    console.error('Error fetching barrel events::', error)
    throw error
  }
}

// layout.tsx
'use client'
import DashboardLayer from '@/components/DashboardLayer'
import Navbar from '@/components/NavBar/Navbar'
import { Button } from '@/components/ui/button'
import {
  getBarrel_EventReasons,
  getBarrel_EventsTypes
} from '@/lib/clientUtils'
import {
  AuthenticatedTemplate,
  UnauthenticatedTemplate
} from '@azure/msal-react'
import Link from 'next/link'
import './dashboard.css'

export default function DashBoardLayout({
  children
}: Readonly<{
  children: React.ReactNode
}>) {
  // Fetching event types and reasons and storing them in local storage
  getBarrel_EventsTypes()
  getBarrel_EventReasons()

  return (
    <section className="relative h-full w-full overflow-y-hidden">

**Comment by wwwdepot** [comment, marked as answer]
Read the docs https://nextjs.org/docs/app/building-your-application/configuring/environment-variables

Pay attention to **Note that dynamic lookups will not be inlined, such as:**

**Comment by mike-rambil** [comment]
@wwwdepot thanks
